protocol-messages.go

```diff
@@ -0,0 +1,380 @@
+package mls
+
+import (
+	"fmt"
+	"github.com/bifurcation/mint/syntax"
+)
+
+////// ClientInitKey
+
+// enum {
+//	invalid(0),
+//	supported_versions(1),
+//	supported_ciphersuites(2),
+//	expiration(3),
+//	(65535)
+// } ExtensionType;
+
+// struct {
+//	ExtensionType extension_type;
+//	opaque extension_data<0..2^16-1>;
+// } Extension;
+
+type ExtensionType uint16
+type Extension struct {
+	ExtensionType ExtensionType
+	ExtensionData []byte `tls:"head=2"`
+}
+
+// struct {
+//	ProtocolVersion supported_version;
+//	opaque client_init_key_id<0..255>;
+//	CipherSuite cipher_suite;
+//	HPKEPublicKey init_key;
+//	Credential credential;
+//	Extension extensions<0..2^16-1>;
+//	opaque signature<0..2^16-1>;
+// } ClientInitKey;
+type ClientInitKey struct {
+	ProtocolVersion uint8
+	CipherSuite     CipherSuite
+	HPKEPublicKey   []byte `tls:"head=2"`
```

Comment:
It would probably be worth defining a simple `HPKEPublicKey` struct to ensure consistent marshal/unmarshal.  Just something like 
```
type HPKEPublicKey struct{
    Data []byte `tls:"head=2"`
}
```

Cf. the `mint` [ExtensionList](https://github.com/bifurcation/mint/blob/master/extensions.go#L32) struct, which may be worth replicating here.

Reply:
That would also make the typing clearer for structs that include HPKEPublicKey fields.